### PR TITLE
spring-web: Akka 2.6 features

### PR DESCRIPTION
**Pull request by dotbg**
References #2499

### CODE REVIEW

**Comment by ennru** [comment]
This changes the API a bit - which is OK for Alpakka 3.0.

But MiMa requires an exclusion rule. See https://github.com/akka/alpakka/blob/master/CONTRIBUTING.md#binary-compatibility-mimahttps://assets.grammarly.com/emoji/v1/1f628.svg

**Comment by dotbg** [comment]
> 
> 
> This changes the API a bit - which is OK for Alpakka 3.0.
> 
> But MiMa requires an exclusion rule. See https://github.com/akka/alpakka/blob/master/CONTRIBUTING.md#binary-compatibility-mima https://assets.grammarly.com/emoji/v1/1f628.svg

@ennru Actually, I can make the API compatible. Should I? May not play well with the `fatalWarnings `, though

File: spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java [hunk 1]
```diff
@@ -25,15 +26,16 @@ public void registerAdapters(ReactiveAdapterRegistry registry) {
...
         akka.stream.javadsl.Source::fromPublisher);
 
     registry.registerReactiveType(
         multiValue(akka.stream.scaladsl.Source.class, akka.stream.scaladsl.Source::empty),
         source ->
             ((akka.stream.scaladsl.Source<?, ?>) source)
-                .runWith(akka.stream.scaladsl.Sink.asPublisher(true), materializer),
+                .runWith(
+                    akka.stream.scaladsl.Sink.asPublisher(true),
+                    Materializer.matFromSystem(system)),
```
Comment (ennru) [inline spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java:38]: Shouldn't this be just `system`?

**Review by ennru — COMMENTED** [review]
Adding back a deprecated `public AkkaStreamsRegistrar(Materializer mat)` which does `mat.system` and a corresponding getter would be enough, right?
Than go for it.

File: spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java (hunk 1 above)
Comment (dotbg) [reply spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java:38]: sadly, I have the compile-time error in that case :( Perhaps, I should do the relevant change in akka-streams.

Comment (ennru) [reply spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java:38]: No there are no changes required. The problem here is that it is using the Scala DSL from Java code. Your explicit use of `matFromSystem` is correct.

Comment (dotbg) [reply spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java:38]: yes, indeed. 
However, the logic does require scala dsl. Hence, my thought on having the corresponding method (accepting `system`, not `materializer`)

Comment (dotbg) [reply spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java:38]: I can wrap the scaladsl.Source with the javadsl.Source.
Besides, the constructor may look as follows 

```java
    /**
     * @deprecated use {@link #AkkaStreamsRegistrar(ActorSystem)}.
     */
  @Deprecated
  public AkkaStreamsRegistrar(Materializer materializer) {
      this(materializer.system());
  }
```

Shall I ?

**Comment by ennru** [comment]
As the signature had to change from `ActorMaterializer` anyway, I just removed it altogether.

**Comment by dotbg** [comment]
Rebased the branch.

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you!

**Comment by dotbg** [comment]
Thank you, @ennru

**Pull request merged**